Jenkinsfile

```diff
@@ -3,57 +3,193 @@
+  agent none
+  parameters {
+    string(
+      name: 'TOOLS_VERSION',
+      defaultValue: '15.0.2',
+      description: 'The tools version to build with (check /projects/tools/ReleasesTools/)'
+    )
   }
   stages {
-    stage('Get view') {
-      steps {
-        xcorePrepareSandbox("${VIEW}", "${REPO}")
+    stage('Standard build and XS1/2 tests') {
+      agent {
+        label 'x86_64&&brew'
       }
-    }
-    stage('Library checks') {
-      steps {
-        xcoreLibraryChecks("${REPO}")
+
+      environment {
+        REPO = 'lib_spi'
+        // VIEW = "${env.JOB_NAME.contains('PR-') ? REPO+'_'+env.CHANGE_TARGET : REPO+'_'+env.BRANCH_NAME}"
+        VIEW = "lib_spi_feature_xs3_support"
       }
-    }
-    stage('Legacy tests') {
-      steps {
-        dir("${REPO}/legacy_tests") {
-          viewEnv() {
-            // Use Pipfile in legacy_tests, not lib_spi/Pipfile
-            installPipfile(true)
-            runPython("./runtests.py --junit-output=${REPO}_tests.xml")
+      options {
+        skipDefaultCheckout()
+      }
+      stages {
+        stage('Get view') {
+          steps {
+            xcorePrepareSandbox("${VIEW}", "${REPO}")
+          }
+        }
+        stage('Library checks') {
+          steps {
+            xcoreLibraryChecks("${REPO}")
+          }
+        }
+        stage('Legacy tests') {
+          steps {
+            dir("${REPO}/legacy_tests") {
+              viewEnv() {
+                // Use Pipfile in legacy_tests, not lib_spi/Pipfile
+                // installPipfile(true)
+                // runPython("./runtests.py --junit-output=${REPO}_tests.xml")
+              }
+            }
+          }
+        }
+
+        stage('Build XCOREAI') {
+          steps {
+            dir("${REPO}") {
+              forAllMatch("examples", "AN*/") { path ->
+                runXmake(path, '', 'XCOREAI=1')
+                dir(path) {
+                  stash name: path.split("/")[-1], includes: 'bin/*, '
+                }
+              }
+              stash name: "reset_xtags", includes: "**/python/reset_xtags.py"
+
+              // Build Tests
+              dir('legacy_tests/') {
+                script {
+                  tests = [
+                    "spi_master_async_multi_client",
+                    "spi_master_async_multi_device",
+                    "spi_master_async_rx_tx",
+                    "spi_master_async_shutdown",
+                    "spi_master_sync_benchmark",
+                    "spi_master_sync_clock_port_sharing",
+                    "spi_master_sync_multi_client",
+                    "spi_master_sync_multi_device",
+                    "spi_master_sync_rx_tx",
+                    "spi_slave_benchmark",
+                    "spi_slave_rx_tx"
+                  ]
+                  tests.each() {
+                    dir(it) {
+                      //Build all test under XS3 to check for build errors
+                      runXmake(".", "", "XCOREAI=1")
```

Comment:
Where does this get run? I can't see if in http://srv-bri-jcim0:8080/blue/organizations/jenkins/XMOS%2Flib_spi/detail/feature%2Fxs3_support/38/pipeline

Reply:
These tests do not get run. However they are built, so at least the compiler checks them. This is mentioned here:

https://github.com/xmos/lib_spi/issues/48

Reply:
Thank you for the explanation.